### PR TITLE
template react-urql-graphql-codegen

### LINKED ISSUE #96: 7a. package iteria-app/lowcode/templates/react-urql-graphql-codegen
Goal: new template as source code for react lowcode codegen. It should be very simple and fully working react web app.

# Template

- [ ] Create React App (no router) with graphql codegen plugin `typescript-urql` under package iteria-app/lowcode/templates/react-urql-graphql-codegen (monorepo) with following files: 

file `src/App.tsx`  is using `useGeneratedQuery()` (from `generated.graphql` by graphql codegen)  and `<ListPlacholder`
```
import { useGeneratedQuery } from 'src/generated'

function App() {
  const [result] = useGeneratedQuery({
    variables: { search }
  })

  if (result.fetching) return <Fetching />;
  if (result.error) return <Error error={error} />;

  return (
    <ListPlaceholder data={result.data.customers} />
  );
}

export default App;
```

file `src/query.graphql`:
```
query generated {
    # ...
}
```

# Script (TBD)
- [ ] script `list-page-codegen.ts` under  `iteria-app/lowcode/templates/react-urql-graphql-codegen/scripts` with following inputs and outputs
input: graphql query string
outputs:
* file `<queryName>.graphql`
* file `<routes>/<entity>/<entity>-container.tsx`
  * replace `useGeneratedQuery` with `'use' + queryName + 'Query'`
  * replace `<ListPlaceholder` with a "real implementation" (component in `<routes>/<entity>/<entity>.tsx`) (TODO attributes)
  * replace `<Loading` with real implementation or remove from AST
  * replace `<Error` with real implementation or remove from AST
  * prettify
* router (ideally automatic registration)
* menu (optional - better most recent - sorted by create/modified timestamp)

### PR DESCRIPTION
**Pull request by chvister**
closes #96 

created grapql template with static query

### CODE REVIEW

File: packages/templates/react-urql-graphql-codegen/src/components/App.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  if (fetching) return <Fetching />;
+  if (error) return <Error error={error} />;
+
+  const handleChange = (value: string) => {
+    setInput(value)
+  }
+
+  return (
+    <>
+      <input
```
Comment (jozef-slezak) [inline packages/templates/react-urql-graphql-codegen/src/components/App.tsx:23]: Please, remove UI components from this template (we will have another template for material-ui ...).

**Review by mat-app — APPROVED** [review]

File: packages/templates/react-urql-graphql-codegen/package.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
+{
+  "name": "react-urql-graphql-codegen",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
```
Comment (jozef-slezak) [inline packages/templates/react-urql-graphql-codegen/package.json:5]: Please vheck once again devDependencies

File: packages/templates/react-urql-graphql-codegen/src/components/App.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { useGeneratedQuery } from '../generated'
+import Fetching from './Fetching'
+import Error from './Error'
+import ListPlaceholder from './ListPlaceholder'
+
+function App() {
+  const [result] = useGeneratedQuery({
+    variables: { search: "%%" }
```
Comment (jozef-slezak) [inline packages/templates/react-urql-graphql-codegen/src/components/App.tsx:8]: Is or possible `variables: {}`?

File: packages/templates/react-urql-graphql-codegen/src/scripts/list-page-codegen.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+
+function listPage(queryInput: string) {
```
Comment (jozef-slezak) [inline packages/templates/react-urql-graphql-codegen/src/scripts/list-page-codegen.ts:2]: What is purpose of this fike?

File: packages/templates/react-urql-graphql-codegen/src/index.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import {
+  createClient,
+  Provider,
+  debugExchange,
+  cacheExchange,
+  fetchExchange
+} from 'urql';
+
+
+const client = createClient({
```
Comment (jozef-slezak) [inline packages/templates/react-urql-graphql-codegen/src/index.tsx:13]: Please use some constant for the endpoint that can be replaced at build time in standard way.

File: packages/templates/react-urql-graphql-codegen/codegen.yml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+overwrite: true
+schema: "https://iteria-app-example01.herokuapp.com/v1/graphql"
```
Comment (jozef-slezak) [inline packages/templates/react-urql-graphql-codegen/codegen.yml:2]: I am thinking how to replace also this endpoint during build time.

File: packages/templates/react-urql-graphql-codegen/src/components/App.tsx (hunk 3 above)
Comment (chvister) [reply packages/templates/react-urql-graphql-codegen/src/components/App.tsx:8]: i changed the query, so it doesn't have to be there  parameter "search"

File: packages/templates/react-urql-graphql-codegen/src/scripts/list-page-codegen.ts (hunk 4 above)
Comment (chvister) [reply packages/templates/react-urql-graphql-codegen/src/scripts/list-page-codegen.ts:2]: deleted

File: packages/templates/react-urql-graphql-codegen/codegen.yml (hunk 6 above)
Comment (chvister) [reply packages/templates/react-urql-graphql-codegen/codegen.yml:2]: hmm can we generate yml file before codegen?

**Comment by chvister** [comment]
@jozef-slezak  I found out that it can be done via .env, but it doesn't work for me yet 

https://www.graphql-code-generator.com/docs/getting-started/codegen-config

image


https://github.com/dotansimha/graphql-code-generator/pull/5442/commits/2502293b061c11767e208a7dabe388451fc96ffc#diff-78171e37f6d36f679d5622c9af2ff55ac970e1a855ad11fd46f50d7a59824ddb

File: packages/templates/react-urql-graphql-codegen/codegen.yml [hunk 7]
```diff
@@ -1,5 +1,5 @@
 overwrite: true
-schema: "https://iteria-app-example01.herokuapp.com/v1/graphql"
+schema: '${REACT_APP_SCHEMA_ENDPOINT:?error with env variable}'
```
Comment (jozef-slezak) [inline packages/templates/react-urql-graphql-codegen/codegen.yml:2]: is it possible to have default here?

**Pull request merged**